### DISCUSSION TITLE
MONAI for non-image data e.g. EEG?

### DISCUSSION THREAD

**Discussion by mandelkow** [Project-MONAI/MONAI#7097] [Q&A]
### Would you recommend MONAI for non-imaging data (e.g. EEG)?
I have used MONAI for imaging data in the past and appreciated in particular the functionality for the randomized sampling of large datasets using SmartCacheDataset e.g.

Now I am considering if MONAI would also bee a good choice for non-imaging data like the EEG.
Would any of the experts have an opinion on this? 
What would be a good starting point for implementing a MONAI dataset with a new data file format? 
So far I have not come across any example code online.

Cheers, Hendrik

**Comment by ebrahimebrahim** [comment]
(Not an expert but throwing in my opinion)

A lot of the strength of MONAI for images comes from the wonderful set of well-documented image transformations.

If you added to MONAI a similarly wonderful set of well-documented EEG signal readers and transformers, I think that would be an amazing contribution!

If you are just wondering whether it's worth using MONAI to start your project, then I'd say why not? It's just pytorch with extra goodies. If you are using pytorch anyway and you want one of those goodies, there's no reason not to import it from monai and use it. For example the smart caching mechanisms for data loading and transformation come to mind. Also metatensors. The "affine" part of a metatensor is not useful to you when working with EEG signals, but you can still benefit from carrying around your own metadata using metatensors.
